teamwork/apps/courses/views/CourseView.py

```diff
@@ -175,13 +179,18 @@ def update_course(request, slug):
     page_description = "Update %s"%(course.name) or "Post a new update"
     title = "Update Course"
 
+    if not request.user.profile.isGT:
+        user_role = Enrollment.objects.filter(user=request.user, course=course).first().role
+    else:
+        user_role = 'GT'
+
     if request.user.profile.isGT:
         pass
-    #if user is not a professor or they did not create course
-    elif not course.creator == request.user:
-        #redirect them to the /course directory with message
-        messages.info(request,'Only Professor can post and update')
-        return HttpResponseRedirect('/course')
+    elif not request.user==course.creator:
```

Comment:
This entire section can just be:
``` 
if user_role == "student":
              create message
              redirect
```
We can delete the 4 extra lines of code. This is repeated in several views.

Reply:
Fixed